adr/ADR-8.md

```diff
@@ -224,6 +224,9 @@ A bucket is a Stream with these properties:
  * Discard Policy is always set to `new`
  * Rollup Headers is always set to `true`
  * Deny Delete is always set to `true`
+ * Allow Direct is allowed to be set by the user.
```

Comment:
I think there is a few other things to consider, ideally KV would work on buckets that suppot Direct but when their environment isnt setup for it.

So we need a few options:

- Be able to select using old style even when the stream has direct (opt-in, off by default)
- Be able to figure out if a bucket supports direct, so maybe `KeyValueStatus` should be updated to include that property
- Finally if a bucket client detects that direct fails with timeout, unauth etc - it should update its internal state to use old style from then on for the length of that client session, this should make migration a bit less onerous and big bang

Reply:
@ripienaar @aricart 
* For direct opt-in, I suppose we could add something to the JetStream/KV/ObjectStore options. Use of direct would be default to auto, but can be turned off?

* For KV, since we allow them to set direct, this makes sense to report directly so they don't have to dig into the backing config.

* direct awareness - this seems like a reasonable client optimization.

Reply:
Background is: At present the KV will detect Direct is set on the stream and use it unconditionally and probably in 2.10 we will once again auto-upgrade all KVish streams to direct.

Anyone who have not updated their JWTs, imports, ACLs and more will then fail - unless the client can fall back to old mode when that happens.

Reply:
After discussion at standup... Allow Direct should be set for the user and it is not modifiable via updating with KV management, only out of band.